File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/di/component/RepositoryComponent.java
package com.kalashnyk.denys.samplemvvmapp.di.component;

import com.kalashnyk.denys.samplemvvmapp.di.module.RepositoryModule;
import com.kalashnyk.denys.samplemvvmapp.di.scope.RepositoryScope;
import com.kalashnyk.denys.samplemvvmapp.repository.AppRepository;

import dagger.Component;

@RepositoryScope
@Component(modules = {RepositoryModule.class}, dependencies = {ApiComponent.class, DaoComponent.class})
public interface RepositoryComponent {
    AppRepository getRepository();
}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/di/component/RepositoryComponent.kt
package com.kalashnyk.denys.kotlinbindingsample.di.component

import com.kalashnyk.denys.kotlinbindingsample.di.module.RepositoryModule
import com.kalashnyk.denys.kotlinbindingsample.di.scope.RepositoryScope
import com.kalashnyk.denys.kotlinbindingsample.repository.AppRepository
import dagger.Component

@RepositoryScope
@Component(modules = [RepositoryModule::class], dependencies = [ApiComponent::class, DatabaseComponent::class])
interface RepositoryComponent {
    val repository: AppRepository
}